### PR TITLE
rename types for v3 and fixup tests

### PR DESCRIPTION
**Pull request by zachgersh**
- also renames FileSnapshotter to FileSnapshot

### CODE REVIEW

**Comment by zachgersh** [comment]
@ZackButcher this also takes care of the rename you suggested during review!

File: pilot/pkg/config/monitor/file_snapshot_test.go [hunk 1]
```diff
@@ -42,8 +42,8 @@ spec:
 `
 
 var routeRuleYAML = `
-apiVersion: config.istio.io/v1alpha2
-kind: V1alpha2RouteRule
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
```
Comment (zachgersh) [inline pilot/pkg/config/monitor/file_snapshot_test.go]: @ldemailly making sure the `kind` was set was the fix.

**Comment by ldemailly** [comment]
unrelated linter failure so merged

**Pull request merged**